libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/DropInNavigationView.kt

```diff
@@ -152,6 +157,32 @@ class DropInNavigationView @JvmOverloads constructor(
     private inline fun <reified T : ViewModel> lazyViewModel(): Lazy<T> = lazy {
         viewModelProvider[T::class.java]
     }
+
+    private fun startTripSession() {
+        val fineLocPermGranted = ActivityCompat.checkSelfPermission(
+            context,
+            Manifest.permission.ACCESS_FINE_LOCATION
+        ) == PackageManager.PERMISSION_GRANTED
+        val coarseLocPermGranted = ActivityCompat.checkSelfPermission(
+            context,
+            Manifest.permission.ACCESS_COARSE_LOCATION
+        ) == PackageManager.PERMISSION_GRANTED
+
+        if (fineLocPermGranted && coarseLocPermGranted) {
+            MapboxNavigationApp.current()?.startTripSession()
```

Comment:
nullable `MapboxNavigationApp.current()?` creates ambiguity. When this logic is part of `MapboxNavigationObserver.onAttached` data streams, it is guaranteed to exist

Reply:
I don't understand what you mean by above comment. 
Do you mean that calling `MapboxNavigationApp.setup()` will not create `MapboxNavigation` instance?
Is the [requested implementation](https://github.com/mapbox/navigation-sdks/issues/1474) wrong?
Please let me know how would you like to do this.

Reply:
What happens if the user accepts user permissions after launching the `DropInNavigationView`. Seems like the tripSession will not start

Suggestion is to make all this "Flowable"

Reply:
If you encapsulate the logic in a view model, it can have tests that are not part of DropInNavigationView

For example, when add the toggle between `startTripSession` and `startReplayTripSession`, this logic is going to be changing. 

Notice the ticket also says "I am not sure if this is how we would want to do it, or abstract it out in the form of a behavior.". The answer to that is, yeah we want to abstract it out :). We have "UIViewModel" not instead of "behavior"